modules/gateway/gateway.go

```diff
@@ -51,94 +53,53 @@ type Gateway struct {
 
 // Address returns the NetAddress of the Gateway.
 func (g *Gateway) Address() modules.NetAddress {
-	counter := g.mu.RLock()
-	defer g.mu.RUnlock(counter)
+	id := g.mu.RLock()
+	defer g.mu.RUnlock(id)
```

Comment:
at some point I decided id was too vague. It's not a big deal but I've started switching to lockID instead. I'll probably go through the whole repo and change all the names to lockID eventually.